public/js/project/canvas/canvas.js

```diff
@@ -79,46 +79,123 @@ function canvas(settings) {
       var aLine = new paper.Path.Line(leftPoint, rightPoint);
       aLine.strokeColor = 'black';
 
-      if (i == num_rectangles_tall / 2) {
+      if (i === num_rectangles_tall / 2) {
         aLine.strokeWidth = 5;
       }
     }
   }
 
   function drawVector(vector) {
-
-  }
-
-  function drawTarget(target) {
-
-  }
+    vectorItem = new Group([
+    new Path(
+    [
+      { x: 250, y: 250 }, 
+      { x: vector[0], y: vector[1]}
+    ]),
+
+      // Arrows
+      new Path([
+        to + arrowVector.rotate(135),
+        to,
+        to + arrowVector.rotate(-135)
+      ])
+    ]);
+
+    vectorItem.strokeColor = 'red';
+    vectorItem.strokeWidth = 5;
+    }
 
   /**
-  * 
-  *
+  * Figure out use case
   */
-
   function checkCollisions(oldVector,newVector,targets) {
 
   }
 
+  function isClose(vector, target, radius) {
+    
+    var vectorX = vector.x,
+        vectorY = vector.y,
+        targetX = target.x,
+        targetY = target.y;
+
+    if (Math.abs(vectorX - targetX) <= radius && Math.abs(vectorY - targetY) <= radius) {
+      return 1;
+    }
+    return 0;
+  }
+  
   /**
   *
   *
   */
-
   function proximity(outputVector, target) {
 
-  }
 
+    if (isClose(targetX, targetY, 500)) {
+      // console.log("Target proximity 500");
+      target.updateColor('#e5e5ff');
+    }
+
+    else if (isClose(targetX, targetY, 450)) {
+      // console.log("Target proximity 450");
+      target.updateColor('#ccccff');
+    }
+
+    else if (isClose(targetX, targetY, 400)) {
+      // console.log("Target proximity 400");
+      target.updateColor('#b2b2ff');
+    }
+
+    else if (isClose(targetX, targetY, 350)) {
+      // console.log("Target proximity 350");
+      target.updateColor('#9999ff');
+    }
+
+    else if (isClose(targetX, targetY, 300)) {
+      // console.log("Target proximity 300");
+      target.updateColor('#7f7fff');
+    }
+
+    else if (isClose(targetX, targetY, 250)) {
+      // console.log("Target proximity 250");
+      target.updateColor('#6666ff');
+    }
+
+    else if (isClose(targetX, targetY, 200)) {
+      // console.log("Target proximity 200");
+      target.updateColor('#4c4cff');
+    }
+
+    else if (isClose(targetX, targetY, 150)) {
+      // console.log("Target proximity 150");
+      target.updateColor('#3232ff');
+    }
+
+    else if (isClose(targetX, targetY, 100)) {
+      // console.log("Target proximity 100");
+      target.updateColor('#1919ff');
+    }
+
+    else if (isClose(targetX, targetY, 50)) {
+      // console.log("Target proximity 50");
+      target.updateColor('#0000ff');
+    }
+  }
 
   /**
-  *
-  *
+  * @PARAM target object as described in target.js
+  * @RETURNS nothing
   */
+  function drawTarget(target) {
```

Comment:
The canvas associated to the paperscript scope associated to that instance of "canvas." A mess, I know...